gadget/update.go

```diff
@@ -20,8 +20,102 @@ package gadget
 
 import (
 	"errors"
+	"fmt"
 )
 
 var (
 	ErrNoUpdate = errors.New("no update needed")
 )
+
+func resolveVolume(old *Info, new *Info) (oldVol, newVol *Volume, err error) {
+	// TODO: support only one volume
+	if len(new.Volumes) != 1 || len(old.Volumes) != 1 {
+		return nil, nil, errors.New("cannot update with more than one volume")
+	}
+
+	var name string
+	for n := range old.Volumes {
+		name = n
+		break
+	}
+	oldV := old.Volumes[name]
+
+	newV, ok := new.Volumes[name]
+	if !ok {
+		return nil, nil, fmt.Errorf("cannot find entry for volume %q in updated gadget info", name)
+	}
+
+	return &oldV, &newV, nil
+}
+
+func isSameOffset(one *Size, two *Size) bool {
+	if one == nil && two == nil {
+		return true
+	}
+	if one != nil && two != nil {
+		return *one == *two
+	}
+	return false
+}
+
+func isSameRelativeOffset(one *RelativeOffset, two *RelativeOffset) bool {
```

Comment:
It seems that theoretically we could have two RelativeOffsets with different contents but with the same meaning (i.e. different offsets to different structures but both resolving to the same absolute offset) but I'm assuming the idea here is to not allow this situation.

Reply:
Yeah, you could come up with such relative offset values that resolve to the same location. Though, as you noticed, it's probably too much hassle to support this, so for simplicity, we expect the unresolved offsets to be literally, the same.